app/src/main/java/com/neu/mobileapplicationdevelopment202430/fragment/PokemonInfoScreen.kt

```diff
@@ -0,0 +1,168 @@
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.navigation.NavController
+import com.bumptech.glide.integration.compose.ExperimentalGlideComposeApi
+import com.bumptech.glide.integration.compose.GlideImage
+import com.neu.mobileapplicationdevelopment202430.pokemon.PokemonItem
+
+@OptIn(ExperimentalGlideComposeApi::class)
+@Composable
+fun PokemonInfoScreen(pokemon: PokemonItem, navController: NavController) {
+    Card (
+        modifier = Modifier
+            .fillMaxWidth(),
+        shape = RoundedCornerShape(32.dp),
+        elevation = CardDefaults.cardElevation(defaultElevation = 2.dp)
+    ) {
+        Column (
+            modifier = Modifier.background(Color(0xFFFAFDF3))
+        ) {
+            Button(
+                colors = ButtonDefaults.buttonColors(
+                    containerColor = Color.White,
+                    contentColor = Color.Black
+                ),
+                onClick = { navController.popBackStack() }) {
+                Text(
+                    text = "Back",
+                    fontSize = 16.sp,
+                    fontWeight = FontWeight.Normal,
+                )
+            }
+
+            Card(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .aspectRatio(1f),
+                shape = RoundedCornerShape(16.dp),
+                elevation = CardDefaults.cardElevation(defaultElevation = 0.dp)
+            ) {
+                GlideImage(
+                    model = pokemon.sprite,
+                    contentDescription = pokemon.name,
+                    contentScale = ContentScale.Crop,
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .aspectRatio(1f)
+                        .clip(RoundedCornerShape(16.dp))
+                )
+            }
+
+            Text(
+                text = pokemon.name,
+                fontSize = 32.sp,
+                fontWeight = FontWeight.Normal,
+                color = Color.Black,
+                textAlign = TextAlign.Center,
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(start = 16.dp, end = 16.dp, top = 4.dp, bottom = 16.dp)
+            )
+
+            Row(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(horizontal = 16.dp, vertical = 4.dp),
+                horizontalArrangement = Arrangement.SpaceBetween
+            ) {
+                Text(
+                    text = "Types:",
+                    fontSize = 24.sp,
+                    fontWeight = FontWeight.Normal,
+                    color = Color.Black
+                )
+
+                Text(
+                    text = pokemon.type1,
+                    fontSize = 24.sp,
+                    fontWeight = FontWeight.Normal,
+                    color = Color.Black
```

Comment:
can we make map that has key - type(String) : value - Color(either Color Object or Hex), and then use that type map to assign font color. For instance 'fire': Color.red, 'water': Color.blue, etc.